### DISCUSSION TITLE
Pagination queries

### DISCUSSION THREAD

**Discussion by tigawanna** [rakkasjs/rakkasjs#95] [Q&A] [answered]
I haven't yet gotten to this use case , but I was wondering how infinite query's would work , given react-query has a useInfiniteQuery which i didn't seem  find in the code , if one were to try and implement it where would one start and what would one look out for

**Comment by cyco130** [comment, marked as answer]
That's true, there's no `useInfiniteQuery` yet.

`react-query` has lots and lots of features. It's not really wise (or feasible) to replicate all of its functionality in Rakkas. For simple cases, you can put pagination parameters into the query key (e.g .`articles/${pageNo}`). For more advanced cases there is a full [react-query integration example](https://github.com/rakkasjs/tanstack-query). Hopefully v0.7 will have a better integration with plugins/recipes.

I wouldn't object if someone came up with a `useInfiniteQuery` implementation that didn't increase the bundle size when not used and that included tests with good coverage  but it's not a priority right now.

In fact, I'm extracting the mini react-query bits into a [separate library](https://github.com/cyco130/good-boy). Maybe it can be a good fit there once its ready.
